crates/viewer/re_dataframe_ui/src/re_table.rs

```diff
@@ -0,0 +1,139 @@
+use crate::re_table_utils::{TableConfig, apply_table_style_fixes, cell_ui, header_ui};
+use egui::{Context, FontSelection, Id, Rangef, RichText, TextWrapMode, Ui, WidgetText};
+use egui_table::{CellInfo, HeaderCellInfo, PrefetchInfo};
+use re_format::format_uint;
+use re_ui::{TableStyle, UiExt as _};
+use std::iter;
+use std::sync::Arc;
+
+/// Wrapper around [`egui_table`] that handles styling, selection, column visibility, row numbers, etc.
+pub struct ReTable<'a> {
+    session_id: Id,
+    inner: &'a mut dyn egui_table::TableDelegate,
+    config: &'a TableConfig,
+    num_rows: u64,
+    table_style: TableStyle,
+    original_style: Arc<egui::Style>,
```

Comment:
the purpose of this needs a docstring, it's not immediately obvious to me what it is exactly.